src/services/date_notes.js

```diff
@@ -62,6 +63,86 @@ function getRootCalendarNote() {
     return rootNote;
 }
 
+function getParentNote(label, dateStr, rootNote) {
+    const pattern = rootNote.getOwnedLabelValue("calendarPattern") || "year/month/day";
```

Comment:
This `calendarPattern` suggests a great deal of flexibility, but what are actually reasonable structures? I can think of just two:

* `year/month/day` (existing)
* `year/week/day` - new

Are there any other? You could imagine e.g. `year/month/week/day`, but I'm not sure how useful that would be. The syntax also allows weird things like `month/year/day` which probably make no sense at all.

In my experience it's often better to provide well-defined options, in this case e.g. `#calendarType=monthly` and `#calendarType=weekly` which force a specific structure.

What do you think?

Reply:
Yes, you're right, using `#calendarType=monthly|weekly` will give a more reasonable journal structure.
Sometimes, I want to omit "year" notes to make journal structure more flatten, and when new year is arrived, I'll move and archive them in another directory and create a new one.

In fact, if I have a chance to redesign the journal function, I would like to make its structure invisible, and the day notes can only be accessible via the calendar widget or apis. Because in the case of contained weekly notes, it's difficult to represent the journal in a tree structure.
But maybe Trilium doesn't want to make journal opaque. So I'll organize the journal by this way: all year notes in "years" folder, all month notes in "month" folder, ... and all day notes in "days" folder, and uses "note-detail-pane" or "right-pane" to display its children. 

What do you think of this idea?

Reply:
> So I'll organize the journal by this way: all year notes in "years" folder, all month notes in "month" folder, ... and all day notes in "days" folder, and uses "note-detail-pane" or "right-pane" to display its children.

But what benefit would this bring? Year-month-day or Year-week-day is a natural hierarchy so it seems fitting to represent it as such in a tree structure.

Reply:
In my opinion, Year-week-day is not as natural as Year-month-day structure.

Weekly note is a little embarrassing when it comes across the end of month or year, but it'd be really useful to be an independent note, rather than the note of the first day of week.

That idea comes from Obsidian's [Periodic Notes](https://github.com/liamcain/obsidian-periodic-notes), which can set the note folder of daily/weekly/monthly/quarterly/yearly notes separately. (BTW, in some companies, they use two weeks as a [sprint](https://www.atlassian.com/agile/scrum/sprints).)

Therefore, I think we can treat the calendar as the database of all kinds of periodic notes, and the hierarchical structure is managed by the calendar widgets or other panels.

Reply:
> In my opinion, Year-week-day is not as natural as Year-month-day structure.

That's why you will be able to choose.

> Weekly note is a little embarrassing when it comes across the end of month or year, but it'd be really useful to be an independent note, rather than the note of the first day of week.

I'm confused. Week should be an independent note, just created in a year-week-day hierarchy. It's a separate note which has (typically 7) children notes.

> Therefore, I think we can treat the calendar as the database of all kinds of periodic notes, and the hierarchical structure is managed by the calendar widgets or other panels. 

I don't think this is a good idea. Besides, since this is a major change, it would break expectations of existing users.

Reply:
Yes, this will be a breaking change.

In summary, you prefer to use `#calendarType=monthly|weekly`, and for the number of weeks, you also prefer to give an option in ISO week or other numbering system. Am I right?

The reason why I want to use that breaking structure is that, I really hope the weekly notes can coexist with monthly notes. Even for `#calendarPattern`, users can use `#calendarPattern=year/month/week/day` to achieve that.

But `#calendarType` is really interesting, it means this attribute can be other values in the future. Anyway, I'm glad to make contribution on the calendar feature.

Reply:
> you prefer to use #calendarType=monthly|weekly

Yes.

> and for the number of weeks, you also prefer to give an option in ISO week or other numbering system. Am I right?

Perhaps in the future, but in the beginning I believe one numbering system is enough. I'm just questioning whether ISO is the one which should be implemented in the beginning.